### ISSUE TITLE
Nullpointer Exception in AutoCompletion on MacOS

### ISSUE THREAD

**Issue by BjoernLoetters** [morganey-lang/Morganey#288]
```
λ> load Exception in thread "main" java.lang.NullPointerException
	at scala.collection.mutable.WrappedArray$ofRef.length(WrappedArray.scala:126)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:32)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
	at scala.collection.immutable.List.flatMap(List.scala:327)
	at me.rexim.morganey.ReplAutocompletion$.topLevelDefinitions$1(ReplAutocompletion.scala:55)
	at me.rexim.morganey.ReplAutocompletion$.autocompleteLoadStatement(ReplAutocompletion.scala:98)
	at me.rexim.morganey.ReplAutocompletion$.complete(ReplAutocompletion.scala:32)
	at me.rexim.morganey.Main$TerminalReplAutocompletion.complete(Main.scala:42)
	at jline.console.ConsoleReader.complete(ConsoleReader.java:3261)
	at jline.console.ConsoleReader.readLine(ConsoleReader.java:2621)
	at jline.console.ConsoleReader.readLine(ConsoleReader.java:2269)
	at jline.console.ConsoleReader.readLine(ConsoleReader.java:2257)
	at me.rexim.morganey.Main$.line$1(Main.scala:56)
	at me.rexim.morganey.Main$.startRepl(Main.scala:58)
	at me.rexim.morganey.Main$.main(Main.scala:91)
	at me.rexim.morganey.Main.main(Main.scala)
```

--

Steps to reproduce the error:
- Start the REPL
- type `load <tab>`

**Comment by rexim** [comment]
@kuchenkruste, thanks for reporting this! Could you please clarify how exactly you start the REPL? Via `sbt run` or via fatjar?

**Comment by keddelzz** [comment]
The REPL was started via the fatjar. He used the version from the master-branch.

**Comment by rexim** [comment]
> The REPL was started via the fatjar. He used the version from the master-branch.

And another condition is there should be no `std/` folder in the current directory. So the full reproduction steps for a developer are:

```
$ cd Morganey
$ sbt assembly
$ cd target
$ java -jar scala-2.11/morganey-assembly-0.0.1.jar
λ> load <tab>
<NPE>
```

I can reproduce that on my Linux too. This is probably caused by the fact that autocompletion uses old file-based mechanism of discovering modules. This issue should be resolved by #257.

---

On the other hand, the application should not fail with NPE just because there is no `std/` folder. Which is a legit defect that can be fixed independently from #257.

**Comment by rexim** [comment]
Let me see what we can do here...

**Comment by BjoernLoetters** [comment]
Just to be clear: The NPE occurred despite there was a `std/` folder relative to the fatjar. Furthermore the "load" command did succeed, only the autocompletion failed.

**Comment by rexim** [comment]
> The NPE occurred despite there was a `std/` folder relative to the fatjar.

That is strange. Did this folder contain anything?

> Furthermore the "load" command did succeed, only the autocompletion failed.

That's an expected behaviour. :) Loading and autocomplete use different module discovery mechanisms until #257 is resolved. It succeeded because it loaded them right from the fatjar. Autocomplete cannot load stuff from jars yet.

**Comment by rexim** [comment]
@kuchenkruste btw, you can checkout the branch from #289 and check if it fixes the issue in your environment. :)

**Comment by rexim** [comment]
@kuchenkruste the fix is merged to master. Please check if the problem is still reproducible for you. Thanks!

**Comment by BjoernLoetters** [comment]
I'll do that asap :-)

**Comment by BjoernLoetters** [comment]
It seems like that the NPE doesn't occur anymore. Sadly the autocompletion still doesn't work either (`load <tab>` simply does nothing). Maybe there is a problem with sym-links? I've moved the fatjar (as well as the `std/` folder) to a separate folder, created a .sh script and linked /usr/local/bin/morganey to it. 
It may be possible that I screwed up the path to the fatjar in the shell script. I'll check this out.
